client/gutenberg/extensions/related-posts/edit.jsx

```diff
@@ -0,0 +1,124 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import classNames from 'classnames';
+import { __ } from '@wordpress/i18n';
+import { Fragment } from '@wordpress/element';
+import { BlockControls, InspectorControls } from '@wordpress/editor';
+import {
+	Button,
+	PanelBody,
+	RangeControl,
+	TextControl,
+	ToggleControl,
+	Toolbar,
+} from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { DEFAULT_POSTS, MAX_POSTS_TO_SHOW } from './constants';
+
+export default ( { attributes, setAttributes } ) => {
+	const {
+		displayContext,
+		displayDate,
+		displayThumbnails,
+		headline,
+		postLayout,
+		postsToShow,
+	} = attributes;
+
+	const layoutControls = [
+		{
+			icon: 'grid-view',
+			title: __( 'Grid View' ),
+			onClick: () => setAttributes( { postLayout: 'grid' } ),
+			isActive: postLayout === 'grid',
+		},
+		{
+			icon: 'list-view',
+			title: __( 'List View' ),
+			onClick: () => setAttributes( { postLayout: 'list' } ),
+			isActive: postLayout === 'list',
+		},
+	];
+
+	const displayPosts =
+		DEFAULT_POSTS.length > postsToShow ? DEFAULT_POSTS.slice( 0, postsToShow ) : DEFAULT_POSTS;
+
+	return (
+		<Fragment>
+			<InspectorControls>
+				<PanelBody title={ __( 'Related Posts Settings' ) }>
+					<TextControl
+						label={ __( 'Headline' ) }
+						value={ headline }
+						onChange={ value => setAttributes( { headline: value } ) }
+					/>
+					<ToggleControl
+						label={ __( 'Display date' ) }
+						checked={ displayDate }
+						onChange={ value => setAttributes( { displayDate: value } ) }
+					/>
+					<ToggleControl
+						label={ __( 'Display thumbnails' ) }
+						checked={ displayThumbnails }
+						onChange={ value => setAttributes( { displayThumbnails: value } ) }
+					/>
+					<ToggleControl
+						label={ __( 'Display context (category or tag)' ) }
+						checked={ displayContext }
+						onChange={ value => setAttributes( { displayContext: value } ) }
+					/>
+					<RangeControl
+						label={ __( 'Number of posts' ) }
+						value={ postsToShow }
+						onChange={ value => setAttributes( { postsToShow: value } ) }
+						min={ 1 }
+						max={ MAX_POSTS_TO_SHOW }
+					/>
+				</PanelBody>
+			</InspectorControls>
+			<BlockControls>
+				<Toolbar controls={ layoutControls } />
+			</BlockControls>
+
+			<div className="related-posts__main">
```

Comment:
Should we rely on generated `className` prop here? Both `edit` and `save` have `className` prop available and it should be `wp-block-a8c-related-posts` (or `wp-block-jetpack-related-posts`).

https://wordpress.org/gutenberg/handbook/block-api/block-edit-save/#classname

Reply:
Good point 👍 

I think the generated `className` weren't there yet when I started working on this block 😉 

I've updated the block to use that class and cleaned up any unnecessary instances in https://github.com/Automattic/wp-calypso/pull/26530/commits/b9e6a6e70fa94f88dc96682bbebb5cbe2c27fb5f